wisp/wisp.php

```diff
@@ -372,11 +394,74 @@ function wisp_CreateAccount(array $params) {
+        // Check if additional ports have been set
+        if(isset($additional_ports) && $additional_ports != '') {
+        
+            // Query all nodes for the given location until we find an available set of ports
+            // Get the list of additional ports to add
+            //$additional_port_list = explode(",", $additional_ports);
+            $additional_port_list = $additional_ports;
+            // Get the server nodes for the specified location_id
+            $nodes = getNodes($params, $location_id);
+            
+            // Get the port allocations for each node at this location and check if there's space for the additional ports
+            if(isset($nodes)) {
+                $alloc_success = false;
+                foreach($nodes as $key => $node_id) {
+                    logModuleCall("WISP-WHMCS", "Checking allocations for node: ".$node_id, "", "");
+                    
+                    // Get all the available allocations for this node
+                    $available_allocations = getAllocations($params,$node_id);
+                    
+                    // Taking our additional allocation requirements and available node allocations, find a combination of available ports.
+                    $final_allocations = findFreePorts($available_allocations, $additional_port_list);
+
+                    if($final_allocations != false && $final_allocations['status'] == true) {
+                        $alloc_success = true;
+                        logModuleCall("WISP-WHMCS", "Successfully found an allocation. Setting primary allocation to ID " . $final_allocations['main_allocation_id'], "", "");
+                        
+                        $serverData['allocation']['default'] = intval($final_allocations['main_allocation_id']);
+                        $serverData['allocation']['additional'] = $final_allocations['additional_allocation_ids'];
+                        
+                        // Update the environment parameters - additional allocations
+                        foreach($final_allocations['additional_allocation_ports'] as $key => $port) {
+                            // If the key given in the config had a value of NONE, don't worry about adding it to the environment parameters.
+                            if(substr( $key, 0, 5 ) !== "NONE_") {
+                                $serverData['environment'][$key] = $port;
+                            }
+                        }
+                        // We successfully found and assigned an available allocation, break and check no more nodes.
+                        break;
+                    }
+                    logModuleCall("WISP-WHMCS", "Failed to find an available allocation on node: ".$node_id, "", "");
+                }
+                if(!$alloc_success) {
+                    // Failure handling logic
+                    if($additional_port_fail_mode = "Stop") {
```

Comment:
assignment instead of comparison, `=` => `==`

Reply:
Although I haven't tested this locally yet, I'd also make sure that the value is indeed `Stop` instead of `stop` (as p sure uppercase version is only a display name).

Reply:
Rookie error on my part, fixed 🤦‍♂️ 
Thanks!